hedera-node/hedera-app/src/main/java/com/hedera/node/app/grpc/HelidonGrpcServerManager.java

```diff
@@ -0,0 +1,207 @@
+        this.services = requireNonNull(services);
+        this.configProvider = requireNonNull(configProvider);
+    }
+
+    @Override
+    public int port() {
+        return server == null ? -1 : server.port();
+    }
+
+    @Override
+    public int tlsPort() {
+        return tlsServer == null ? -1 : tlsServer.port();
+    }
+
+    @Override
+    public void start() {
+        if (server != null || tlsServer != null) {
+            logger.error("Cannot start gRPC servers, they have already been started!");
+            throw new IllegalStateException("Server already started");
+        }
+
+        logger.info("Starting gRPC servers");
+        final var grpcConfig = configProvider.getConfiguration().getConfigData(GrpcConfig.class);
+        final var port = grpcConfig.port();
+        final var tlsPort = grpcConfig.tlsPort();
+
+        if (invalidPort(port)) {
+            throw new IllegalArgumentException("Invalid port number: " + port);
+        }
+
+        if (invalidPort(tlsPort)) {
+            throw new IllegalArgumentException("Invalid tls port number: " + tlsPort);
+        }
+
+        // Setup the GRPC Routing, such that all grpc services are registered
+        final var grpcRoutingBuilder = GrpcRouting.builder();
+        services.forEach(grpcRoutingBuilder::register);
+
+        // Attempt to start the servers
+        final var nettyConfig = configProvider.getConfiguration().getConfigData(NettyConfig.class);
+        final var startRetries = nettyConfig.startRetries();
+        final var startRetryIntervalMs = nettyConfig.startRetryIntervalMs();
+
+        try {
+            logger.debug("Starting Helidon gRPC server on port {}", port);
+            server = GrpcServer.create(GrpcServerConfiguration.builder().port(port), grpcRoutingBuilder);
+            startServerWithRetry(server, startRetries, startRetryIntervalMs);
+            logger.debug("Helidon gRPC server listening on port {}", server.port());
+
+            logger.debug("Starting Helidon TLS gRPC server on port {}", tlsPort);
+            tlsServer = GrpcServer.create(
+                    GrpcServerConfiguration.builder()
+                            .port(port)
+                            .tlsConfig(GrpcTlsDescriptor.builder()
+                                    .enabled(true)
+                                    .tlsCert(Resource.create(Path.of(nettyConfig.tlsCrtPath())))
+                                    .tlsKey(Resource.create(Path.of(nettyConfig.tlsKeyPath())))
+                                    .build())
+                            .build(),
+                    grpcRoutingBuilder);
+            startServerWithRetry(tlsServer, startRetries, startRetryIntervalMs);
+            logger.debug("Helidon TLS gRPC server listening on port {}", tlsServer.port());
+        } catch (ResourceException e) {
+            tlsServer = null;
+            logger.warn("Could not start TLS server, will continue without it: {}", e.getMessage());
```

Comment:
Maybe I am missing something, but in my understanding, this catch-block is also executed when the regular server cannot be started. Or are we throwing a different `Exception` in this case?